src/gui/windows/chat.zig

```diff
@@ -182,15 +182,15 @@ pub fn loadPreviousHistoryEntry(_: usize) void {
 	if(messageHistory.isDuplicate(input.currentString.items)) {} else {
 		messageHistory.pushUp(main.globalAllocator.dupe(u8, input.currentString.items));
 	}
-	const msg = messageHistory.down.peekFront() orelse "";
+	const msg = messageHistory.down.peekBack() orelse "";
 	input.setString(msg);
 }
 
 pub fn onClose() void {
 	while(history.popOrNull()) |label| {
 		label.deinit();
 	}
-	while(messageQueue.dequeue()) |msg| {
+	while(messageQueue.pop()) |msg| {
```

Comment:
This code is direction aware and it is important if we are popping front or back, it should use descriptive function name.